components/patient-data/impl/src/main/java/org/phenotips/data/internal/PhenoTipsPatient.java

```diff
@@ -220,6 +221,12 @@ public JSONObject toJSON(Collection<String> selectedFields)
     @Override
     public void updateFromJSON(JSONObject json)
     {
+        updateFromJSON(json, PatientWritePolicy.UPDATE);
+    }
+
+    @Override
+    public void updateFromJSON(JSONObject json, PatientWritePolicy policy)
```

Comment:
I only realized this while working on the "improve entities" issue: `updateFromJSON()` is actually a `PrimaryEntity` method, should we have a more generic `EntityWritePolicy` instead of `PatientWritePolicy` and correspondingly have `updateFromJSON(JSON, Policy)` be a `PrimaryEntity` method? The default implementation may throw an "unimplemented exception" for all policies which are different from current default to make it compatible with all other current entities which do not support an "update" yet

Reply:
Update: discussed this with @a-gvo, she thinks this is a good idea. I will implement that as part of the update entities branch. Note, since Entity update will be in 1.4m3, 1.4m2 will have this API which will change again for 1.4m3, hope that does not cause any problems (@sdumitriu?)

Reply:
This should indeed be part of the entities API. OK to have this temporary API in 1.4M2.